core/primitives/src/challenge.rs

```diff
@@ -10,8 +10,19 @@ use near_crypto::Signature;
 pub type StateItem = std::sync::Arc<[u8]>;
 
 #[derive(BorshSerialize, BorshDeserialize, serde::Serialize, Debug, Clone, Eq, PartialEq)]
-pub struct PartialState(pub Vec<StateItem>);
+/// TODO (#8984): consider supporting format containing trie nodes only for
+/// state part boundaries and storing state items for state part range.
+pub enum PartialState {
+    /// State represented by the trie nodes.
+    Nodes(Vec<StateItem>),
```

Comment:
`Nodes` is confusing, because it contains both Trie Nodes and Values referenced by those nodes.

Reply:
How about `TrieItem`s?
I really want to resolve this confusion in the whole codebase and call these entities "items" instead of "nodes" or "nodes-or-values"

Reply:
`TrieItem` sounds good.